Change the wording in `std::fmt::Write::write_str`

tbu- opened the pull request:
Refer to the error instead of expanding its name.

tbu- committed:
Change the wording in `std::fmt::Write::write_str`

Refer to the error instead of expanding its name.

m-ou-se commented:
The full path `std::fmt::Error` seems useful to show, to distinguish it from `std::io::Error` (which is used by the very similar `std::io::Write` trait).

tbu- commented:
> The full path `std::fmt::Error` seems useful to show, to distinguish it from `std::io::Error` (which is used by the very similar `std::io::Write` trait).

Added the path to the line above.

dtolnay commented on library/core/src/fmt/mod.rs:
```diff
@@ -112,9 +112,9 @@ pub trait Write {
     /// # Errors
     ///
-    /// This function will return an instance of [`Error`] on error.
+    /// This function will return an instance of [`std::fmt::Error`] on error.
```
This should unbreak the doc build.

```suggestion
    /// This function will return an instance of [`std::fmt::Error`][Error] on error.
```

dtolnay committed:
Fix "unresolved link to std::fmt::Error"

error: unresolved link to `std::fmt::Error`
       --> library/core/src/fmt/mod.rs:115:52
        |
    115 |     /// This function will return an instance of [`std::fmt::Error`] on error.
        |
        |
        = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`

dtolnay reviewed (approved):
Thanks!

dtolnay commented:
@bors r+ rollup

bors commented:
:pushpin: Commit e2f70324089a2eb75d52552c058170e37d0064f0 has been approved by `dtolnay`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=dtolnay e2f70324089a2eb75d52552c058170e37d0064f0 -->
<!-- homu: {"type":"Approved","sha":"e2f70324089a2eb75d52552c058170e37d0064f0","approver":"dtolnay","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
